lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -579,89 +579,96 @@ def cmd_services(*args)
-        print_line "  -p <port1,port2>  Search for a list of ports"
-        print_line "  -r <protocol>     Only show [tcp|udp] services"
-        print_line "  -u,--up           Only show services which are up"
-        print_line "  -o <file>         Send output to a file in csv format"
-        print_line "  -O <column>       Order rows by specified column number"
-        print_line "  -R,--rhosts       Set RHOSTS from the results of the search"
-        print_line "  -S,--search       Search string to filter by"
-        print_line
-        print_line "Available columns: #{default_columns.join(", ")}"
-        print_line
-        return
-      else
-        # Anything that wasn't an option is a host to search for
-        unless (arg_host_range(arg, host_ranges))
+          col_search = list.strip().split(",")
+          col_search.each { |c|
+            if not default_columns.include? c
+              print_error("Invalid column list. Possible values are (#{default_columns.join("|")})")
+              return
+            end
+          }
+        when '-p'
+          unless (arg_port_range(args.shift, port_ranges, true))
+            return
+          end
+        when '-r'
+          proto = args.shift
+          if (!proto)
+            print_status("Invalid protocol")
+            return
+          end
+          proto = proto.strip
+          opts[:proto] = proto
+        when '-s'
+          namelist = args.shift
+          if (!namelist)
+            print_error("Invalid name list")
+            return
+          end
+          names = namelist.strip().split(",")
+          opts[:name] = names
+        when '-o'
+          output_file = args.shift
+          if (!output_file)
+            print_error("Invalid output filename")
+            return
+          end
+          output_file = ::File.expand_path(output_file)
+        when '-O'
+          if (order_by = args.shift.to_i - 1) < 0
+            print_error('Please specify a column number starting from 1')
+            return
+          end
+        when '-R', '--rhosts'
+          set_rhosts = true
+        when '-S', '--search'
+          search_term = args.shift
+          opts[:search_term] = search_term
+        when '-h','--help'
+          print_line
+          print_line "Usage: services [-h] [-u] [-a] [-r <proto>] [-p <port1,port2>] [-s <name1,name2>] [-o <filename>] [addr1 addr2 ...]"
+          print_line
+          print_line "  -a,--add          Add the services instead of searching"
+          print_line "  -d,--delete       Delete the services instead of searching"
+          print_line "  -c <col1,col2>    Only show the given columns"
+          print_line "  -h,--help         Show this help information"
+          print_line "  -s <name>         Name of the service to add"
+          print_line "  -p <port>         Port number of the service being added"
+          print_line "  -r <protocol>     Only show [tcp|udp] services"
```

Comment:
The description should be more generic given this parameter may be used for adding a new service or filtering

Reply:
If `-r` is only used when adding a service the description should be updated